### PR TITLE
1086 - Show answer when displaying completed activity questions

### PR DESCRIPTION
**Pull request by EldadZZipori**
## Task Description
Right now, when we view an activity node that was already completed, we still show the activity or answer type selection screen. Instead of that, we want to display the answer they gave like the answer node with an edit button that would then take the user to that screen. This should be done PER question, so the buttons to navigate between questions should still be there (but still we will show the answer if they have already completed it).

## Changes
Add a small state machine to activity media with three states
```
const states = {
  ACTIVITY: "activity",
  ANSWER: "answer",
  COMPLETIONSCREEN: "completion-screen",
}
```

Moved the AnswerMedia component to be a part of ActivityMedia
## Concept

| Initial Type | Question Has Answer | State | Edit Answer Button | Show Last Answer Button |
| --- | --- | --- | --- | --- |
| activity | ❎ | activity  | ❎ | ❎ |
| activity | ✅  | answer  | ✅ | ❎ |
| activity | ✅  | activity  | ❎ | ✅ |
| answer | ❎  | answer  | ✅ | ❎ |
| answer | ✅ | answer  | ✅ | ❎ |
| answer | ✅ | activity  | ❎ | ✅ |

* State will never be "answer" if Initial Type is "activity" and Question Has Answer is false

## Issue Linkage
Closes (issue #1086 )
## PR Dependency
Depends on: #1032 
## Automated Testing

1. Refactor all activities test to work with the new functionality

### CODE REVIEW

**Comment by Frozemint** [comment]
The functionality of the feature looks fine. Could the interface be tweaked slightly so that it's more obvious to the user that they're looking at their previously entered answer? Right now it's kind of unclear:

image

**Comment by Frozemint** [comment]
Thanks Eldad, there is a small typo with the show previous answer button, but otherwise all the core functionality as specified works and I'm happy to have the PR merged; we can worry about making more improvements to the previous answer UI later. 
image

**Review by Frozemint — APPROVED** [review]

File: templates/vue/src/components/Lightbox/media/ActivityMedia/index.vue [hunk 1]
```diff
@@ -17,30 +20,58 @@
...
+          variant="info"
+          style="margin-right:auto;"
+          @click="state = 'answer'"
+        >
+          Show previos answers
+        </b-button>
+        <b-button
+          v-else-if="state === 'answer'"
+          variant="info"
+          style="margin-right:auto;"
```
Comment (wynnset) [inline templates/vue/src/components/Lightbox/media/ActivityMedia/index.vue:50]:
```suggestion
          class="mr-auto"
```

File: templates/vue/src/components/Lightbox/media/ActivityMedia/index.vue [hunk 2]
```diff
@@ -50,12 +81,20 @@ import client from "@/services/TapestryAPI"
 import Question from "./Question"
 import CompletionScreen from "./CompletionScreen"
 import { mapActions, mapGetters } from "vuex"
+import AnswerMedia from "./AnswerMedia.vue"
+
+const states = {
+  ACTIVITY: "activity",
+  ANSWER: "answer",
+  COMPLETIONSCREEN: "completion-screen",
```
Comment (wynnset) [inline templates/vue/src/components/Lightbox/media/ActivityMedia/index.vue:89]: For better readability, please use underscore to separate the words here

File: templates/vue/src/components/Lightbox/media/ActivityMedia/index.vue [hunk 3]
```diff
@@ -71,20 +110,35 @@ export default {
...
+      state: "",
     }
   },
   computed: {
-    ...mapGetters(["getAnswers"]),
+    ...mapGetters(["getAnswers", "getQuestion", "getNode"]),
+    questionNode() {
+      return this.initialType === states.ACTIVITY
+        ? this.node
+        : this.getNode(this.node.typeData.activityId)
```
Comment (wynnset) [inline templates/vue/src/components/Lightbox/media/ActivityMedia/index.vue:129]: Can you remind me what an activityId is exactly? I do not remember having this in the data structure. I thought we only have node Ids that are for each activity.

File: templates/vue/src/components/Lightbox/media/ActivityMedia/index.vue [hunk 4]
```diff
@@ -155,7 +252,10 @@ export default {
     },
     close() {
       client.recordAnalyticsEvent("user", "close", "activity", this.node.id)
-      this.$emit("close")
+      if (this.initialType === "activity") this.$emit("close")
```
Comment (wynnset) [inline templates/vue/src/components/Lightbox/media/ActivityMedia/index.vue:255]: Please use curly brackets for clarity and consistency.

File: templates/vue/src/components/Lightbox/media/ActivityMedia/index.vue [hunk 5]
```diff
@@ -113,40 +204,46 @@ export default {
...
-      const progress = numberCompleted / this.node.typeData.activity.questions.length
-      this.updateNodeProgress({ id: this.node.id, progress }).then(() => {
-        if (progress === 1) {
-          this.$emit("complete")
+          question.completed = true
         }
       })
     },
+    handleComplete(sender) {
+      if (sender === states.ACTIVITY) {
```
Comment (wynnset) [inline templates/vue/src/components/Lightbox/media/ActivityMedia/index.vue:217]: I find it a bit confusing to use the states for senders as well, even though they match in their string values.

**Review by wynnset — CHANGES_REQUESTED** [review]
This works great! Just looking at the code and had a few comments and need some clarifications so I can finish my review. In the meantime, feel free to address these comments.

**Review by wynnset — APPROVED** [review]

**Pull request merged**